chainofresponsibilty/src/chainofresponsibilty/handlers/Handler.java

```diff
@@ -0,0 +1,19 @@
+package chainofresponsibilty.handlers;
+
+public abstract class Handler {
+    private Handler next;
+
+    public Handler setNextHandler(Handler next) {
```

Comment:
use constructor instead of setNext,
also let me know later, why should we use constructors instead of setNext here.